### PR TITLE
[4.0] Add versioning traits to frontend controllers

### PR DESCRIPTION
**Pull request by SharkyKZ**
Fixes https://github.com/joomla/joomla-cms/issues/30002,

### Summary of Changes

Adds missing trait to frontend controllers.

### Testing Instructions

Code review or create an article or a contact.
Edit to create some versions.
Edit in frontend. Click `Versions` button.
In the modal select a version and lick `Restore`.

### Actual result BEFORE applying this Pull Request

Redirected to front page.

### Expected result AFTER applying this Pull Request

Version restored.

### Documentation Changes Required

No.

### CODE REVIEW

File: components/com_content/src/Controller/ArticleController.php [hunk 1]
```diff
@@ -17,6 +17,7 @@
 use Joomla\CMS\Language\Text;
 use Joomla\CMS\MVC\Controller\FormController;
 use Joomla\CMS\Router\Route;
+use Joomla\CMS\Versioning\VersionableControllerTrait;
 use Joomla\CMS\Uri\Uri;
```
Comment (Quy) [inline components/com_content/src/Controller/ArticleController.php]:
```suggestion
use Joomla\CMS\Uri\Uri;
use Joomla\CMS\Versioning\VersionableControllerTrait;
```

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 2642aca5f929d1925e1be26061fa99e7dfd953fb<br /><br />Tested successfully for the behaviour in the related issue. The article edit views remains and the correct message is displayed. But the article is not restored. As this is the same in backend, it is another issue.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30004">issues.joomla.org/tracker/joomla-cms/30004</a>.</sub>

**Comment by ReLater** [comment]
I have tested unsuccessfully: Please see https://github.com/joomla/joomla-cms/issues/30006#issuecomment-653942943

**Comment by wilsonge** [comment]
Thanks!

**Comment by wilsonge** [comment]
Merged this because either way this is the correct fix for the frontend. There's clearly a more general problem that needs to be fixed across both interfaces

**Pull request merged**